### DISCUSSION TITLE
Is there any way to export readonly html?

### DISCUSSION THREAD

**Discussion by RandyLiu6410** [react-page/react-page#1068] [Q&A] [answered]
I have used `ref.current.innerHtml` to export html of the preview page. But the output html still contains some features of react-page editor like `<div class="react-page-controls-mode-toggle-control-group" ... />`. Yes I can extract the element `<div class="react-page-editable react-page-editable-mode-preview" ... />` as my output. But it still contains `<div class="react-page-cell-handle react-page-cell-handle-drag-enabled" ...>`. I still can set the .css to display .react-page-cell-handle.eact-page-cell-handle-drag-enabled as `none`. But I still wondering if there another perfect way to export html of the editor like official function of react-page/editor package. 

The purpose of exporting html is I am using react-page/editor to design the content that will shows on e-mail and react-native mobile view.

**Comment by macrozone** [comment, marked as answer]
You can set `readOnly`: https://react-page.github.io/docs/#/quick-start?id=viewing

this will only contain the components needed to render the content. You can denn use 

```
import ReactDOMServer from 'react-dom/server';

ReactDOMServer.renderToString(<Editor value={value} readOnly cellPlugins={cellPlugins} />)

```




```
